.gitignore

```diff
@@ -0,0 +1 @@
+*~
```

Comment:
I don't really love marking this since it is an editor specific ignore. Would love to purge HSE's gitignore. I would rather people use their personal gitignore files or the repo-specific `.git/info/exclude`.